espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/state/InputStateTable.java

```diff
@@ -0,0 +1,26 @@
+package io.appium.espressoserver.lib.helpers.w3c.state;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Keep the state of all active input sources
+ *
+ * (defined in 17.1 of spec)
+ */
+public class InputStateTable {
+
+    private Map<String, InputStateInterface> stateTable = new HashMap<>();
```

Comment:
`final`

Also, I'd prefer to use ImmutableMapBuilder

Reply:
Do you have an example? I couldn't find ImmutableMap.Builder or ImmutableMapBuilder in the libraries.